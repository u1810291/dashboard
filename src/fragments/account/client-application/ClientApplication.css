:local(.copyButtonColumn) {
  width: var(--mgi-spacing);
}

:local(.headerColumn) {
  width: 180px;
}
