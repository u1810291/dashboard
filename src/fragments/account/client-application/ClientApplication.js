import React from 'react'
import { FormattedMessage } from 'react-intl'
import Card from 'components/card'
import Button from 'components/button'
import Icons from 'components/icons'
import Items from 'components/items'
import { createOverlay, closeOverlay } from 'components/overlay'
import confirm from 'components/confirm'
import { copyToClipboard } from 'components/clipboard'
import NewWebhookModal from '../new-webhook-modal'
import CSS from './ClientApplication.module.css'

function CopyToClipboardCell({ text }) {
  return (
    <td className="text-right" onClick={() => copyToClipboard(text)}>
      <Button buttonStyle="invisible">
        <Icons.Clipboard className="svg-primary" />
      </Button>
    </td>
  )
}

export default function ClientApplication({
  application = {},
  webhooks = [],
  subscribeToWebhook = () => {},
  deleteWebhook = () => {}
}) {
  const permalink = `${process.env.REACT_APP_SIGNUP_URL}?merchantToken=${
    application.clientId
  }&metadata={}`

  const handleAddNewWebhook = () => {
    createOverlay(
      <NewWebhookModal onSave={subscribeToWebhook} onClose={closeOverlay} />
    )
  }

  const handleDeleteWebhook = id => {
    confirm(
      <FormattedMessage id="fragments.client_application.confirm_delete_webhook" />
    ).then(() => deleteWebhook(id))
  }

  return (
    <Card>
      <Items flow="row">
        <h3 className="text-active">
          <FormattedMessage id="fragments.client_application.production_title" />
        </h3>
        <hr />
        <table className="mgi-table">
          <colgroup>
            <col className={CSS.headerColumn} />
            <col />
            <col className={CSS.copyButtonColumn} />
          </colgroup>
          <tbody>
            <tr>
              <th>
                <FormattedMessage id="fragments.client_application.client_id" />
              </th>
              <td className="text-secondary">{application.clientId}</td>
              <CopyToClipboardCell text={application.clientId} />
            </tr>
            <tr>
              <th>
                <FormattedMessage id="fragments.client_application.client_secret" />
              </th>
              <td className="text-secondary">{application.clientSecret}</td>
              <CopyToClipboardCell text={application.clientSecret} />
            </tr>
            <tr>
              <th>
                <FormattedMessage id="fragments.client_application.permalink" />
              </th>
              <td className="text-secondary">{permalink}</td>
              <CopyToClipboardCell text={permalink} />
            </tr>
            {webhooks.map((webhook, index) => (
              <tr key={webhook.id}>
                {index === 0 && (
                  <th rowSpan={webhooks.length}>
                    <FormattedMessage id="fragments.client_application.webhook" />
                  </th>
                )}
                <td>
                  <Items templateColumns="minmax(auto, 100%) auto">
                    <code>{webhook.url}</code>
                    <Button
                      buttonStyle="invisible"
                      onClick={() => handleDeleteWebhook(webhook.id)}
                    >
                      <Icons.TrashBin className="svg-error" />
                    </Button>
                  </Items>
                </td>
                <CopyToClipboardCell text={webhook.url} />
              </tr>
            ))}

            {webhooks.length === 0 && (
              <tr>
                <th>
                  <FormattedMessage id="fragments.client_application.webhook" />
                </th>
                <td>
                  <Button buttonStyle="primary" onClick={handleAddNewWebhook}>
                    <FormattedMessage id="fragments.client_application.add_webhook" />
                  </Button>
                </td>
              </tr>
            )}
          </tbody>
        </table>
      </Items>
    </Card>
  )
}
