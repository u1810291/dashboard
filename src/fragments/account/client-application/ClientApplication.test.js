import React from 'react'
import { shallow } from 'enzyme'
import ClientApplication from '.'

describe('ClientApplication', () => {
  const application = {
    clientId: '5c5c929014cb881f5730271e',
    clientSecret: '80ZR35QMCRUFD9XC9AQ377WEMBKCY4X0'
  }
  const webhooks = [
    {
      id: '1',
      url: 'https://mati-callbacks-demo.firebaseio.com/webhooks.json'
    },
    {
      id: '2',
      url: 'https://en860xcg7be8q.x.pipedream.net/'
    }
  ]

  test('no webhooks', () => {
    const component = shallow(<ClientApplication application={application} webhooks={[]} />)
    expect(component).toMatchSnapshot()
  })

  test('one webhook', () => {
    const component = shallow(
      <ClientApplication application={application} webhooks={webhooks.slice(0, 1)} />
    )
    expect(component).toMatchSnapshot()
  })

  test('two webhooks', () => {
    const component = shallow(
      <ClientApplication application={application} webhooks={webhooks.slice(0, 1)} />
    )
    expect(component).toMatchSnapshot()
  })
})
