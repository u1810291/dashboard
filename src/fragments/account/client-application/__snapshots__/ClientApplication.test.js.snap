// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientApplication no webhooks 1`] = `
<Panel>
  <PanelBody>
    <Sections
      withBorder={true}
    >
      <h3
        className="text-active"
      >
        <FormattedMessage
          id="fragments.client_application.production_title"
          values={Object {}}
        />
      </h3>
      <table
        className="mgi-table"
      >
        <colgroup>
          <col />
          <col />
          <col />
        </colgroup>
        <tbody>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.client_id"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              5c5c929014cb881f5730271e
            </td>
            <CopyToClipboardCell
              text="5c5c929014cb881f5730271e"
            />
          </tr>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.client_secret"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              80ZR35QMCRUFD9XC9AQ377WEMBKCY4X0
            </td>
            <CopyToClipboardCell
              text="80ZR35QMCRUFD9XC9AQ377WEMBKCY4X0"
            />
          </tr>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.permalink"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              https://signup.getmati.com?merchantToken=5c5c929014cb881f5730271e&metadata={}
            </td>
            <CopyToClipboardCell
              text="https://signup.getmati.com?merchantToken=5c5c929014cb881f5730271e&metadata={}"
            />
          </tr>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.webhook"
                values={Object {}}
              />
            </th>
            <td>
              <Button
                buttonStyle="primary"
                onClick={[Function]}
              >
                <FormattedMessage
                  id="fragments.client_application.add_webhook"
                  values={Object {}}
                />
              </Button>
            </td>
          </tr>
        </tbody>
      </table>
    </Sections>
  </PanelBody>
</Panel>
`;

exports[`ClientApplication one webhook 1`] = `
<Panel>
  <PanelBody>
    <Sections
      withBorder={true}
    >
      <h3
        className="text-active"
      >
        <FormattedMessage
          id="fragments.client_application.production_title"
          values={Object {}}
        />
      </h3>
      <table
        className="mgi-table"
      >
        <colgroup>
          <col />
          <col />
          <col />
        </colgroup>
        <tbody>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.client_id"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              5c5c929014cb881f5730271e
            </td>
            <CopyToClipboardCell
              text="5c5c929014cb881f5730271e"
            />
          </tr>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.client_secret"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              80ZR35QMCRUFD9XC9AQ377WEMBKCY4X0
            </td>
            <CopyToClipboardCell
              text="80ZR35QMCRUFD9XC9AQ377WEMBKCY4X0"
            />
          </tr>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.permalink"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              https://signup.getmati.com?merchantToken=5c5c929014cb881f5730271e&metadata={}
            </td>
            <CopyToClipboardCell
              text="https://signup.getmati.com?merchantToken=5c5c929014cb881f5730271e&metadata={}"
            />
          </tr>
          <tr
            key="1"
          >
            <th
              rowSpan={1}
            >
              <FormattedMessage
                id="fragments.client_application.webhook"
                values={Object {}}
              />
            </th>
            <td
              className="mgi-items"
            >
              <code
                className="mgi-items--grow"
              >
                https://mati-callbacks-demo.firebaseio.com/webhooks.json
              </code>
              <Button
                buttonStyle="invisible"
                onClick={[Function]}
              >
                <trashbin.svg
                  className="svg-error"
                />
              </Button>
            </td>
            <CopyToClipboardCell
              text="https://mati-callbacks-demo.firebaseio.com/webhooks.json"
            />
          </tr>
        </tbody>
      </table>
    </Sections>
  </PanelBody>
</Panel>
`;

exports[`ClientApplication two webhooks 1`] = `
<Panel>
  <PanelBody>
    <Sections
      withBorder={true}
    >
      <h3
        className="text-active"
      >
        <FormattedMessage
          id="fragments.client_application.production_title"
          values={Object {}}
        />
      </h3>
      <table
        className="mgi-table"
      >
        <colgroup>
          <col />
          <col />
          <col />
        </colgroup>
        <tbody>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.client_id"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              5c5c929014cb881f5730271e
            </td>
            <CopyToClipboardCell
              text="5c5c929014cb881f5730271e"
            />
          </tr>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.client_secret"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              80ZR35QMCRUFD9XC9AQ377WEMBKCY4X0
            </td>
            <CopyToClipboardCell
              text="80ZR35QMCRUFD9XC9AQ377WEMBKCY4X0"
            />
          </tr>
          <tr>
            <th>
              <FormattedMessage
                id="fragments.client_application.permalink"
                values={Object {}}
              />
            </th>
            <td
              className="text-secondary"
            >
              https://signup.getmati.com?merchantToken=5c5c929014cb881f5730271e&metadata={}
            </td>
            <CopyToClipboardCell
              text="https://signup.getmati.com?merchantToken=5c5c929014cb881f5730271e&metadata={}"
            />
          </tr>
          <tr
            key="1"
          >
            <th
              rowSpan={1}
            >
              <FormattedMessage
                id="fragments.client_application.webhook"
                values={Object {}}
              />
            </th>
            <td
              className="mgi-items"
            >
              <code
                className="mgi-items--grow"
              >
                https://mati-callbacks-demo.firebaseio.com/webhooks.json
              </code>
              <Button
                buttonStyle="invisible"
                onClick={[Function]}
              >
                <trashbin.svg
                  className="svg-error"
                />
              </Button>
            </td>
            <CopyToClipboardCell
              text="https://mati-callbacks-demo.firebaseio.com/webhooks.json"
            />
          </tr>
        </tbody>
      </table>
    </Sections>
  </PanelBody>
</Panel>
`;
